### PR TITLE
Create template for documentation issues

### LINKED ISSUE #1470: Create template for documentation issues
**Is your feature request related to a problem? Please describe.**
On https://github.com/nest/nest-simulator/issues/new/choose we currently have templates for bugs, feature requests and vulnerabilities, but not for errors or weaknesses related to documentation. This is problematic, since the other templates do not fit documentation very well.

**Describe the solution you'd like**
Create a template tailored to documentation-related issues.

### PR DESCRIPTION
**Pull request by sarakonradi**
This PR creates a template for documentation-related issues. 

The `documentation_improvement.md` asks the user to: 
- Describe errors or weaknesses in the documentation
- Suggest ways to improve the documentation
- Add supporting materials, desktop/environment details and context (if applicable)

It fixes #1470.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/documentation_improvement.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+name: Documentation improvement
+about: Request a documentation update
+title: ''
+labels: ''
```
Comment (terhorstd) [inline .github/ISSUE_TEMPLATE/documentation_improvement.md]: Could this already be
```suggestion
labels: 'C: Documentation'
```

File: .github/ISSUE_TEMPLATE/documentation_improvement.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+assignees: ''
+
+---
+
+**Describe the flaw**
+A clear description of what the error or weakness in the documentation is.
+
```
Comment (terhorstd) [inline .github/ISSUE_TEMPLATE/documentation_improvement.md]:
```suggestion
A clear description of what the error or weakness in the documentation is.
Give a filename in the repository or a link to the page you are referring to, if applicable.
```

**Comment by sarakonradi** [comment]
@terhorstd Good idea! The label syntax should work fine. I checked, and there are many other projects on GitHub that have successfully used that. 

Perhaps we could apply that syntax to our other issue templates too in a different PR? (e.g. ``T: Bug`` for a bug report)

**Comment by sarakonradi** [comment]
It would be great if @heplesser could have a look at this too.

File: .github/ISSUE_TEMPLATE/documentation_improvement.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+**Suggest an improvement**
+A clear description of what you want to change or add to the documentation.
+
+**Supporting materials**
+If applicable, add links or screenshots to help explain the flaw.
+
+**Desktop/Environment**
```
Comment (heplesser) [inline .github/ISSUE_TEMPLATE/documentation_improvement.md:19]: How much sense does this section make for documentation?

File: .github/ISSUE_TEMPLATE/documentation_improvement.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+---
+name: Documentation improvement
+about: Request a documentation update
```
Comment (heplesser) [inline .github/ISSUE_TEMPLATE/documentation_improvement.md:3]: "update" -> "improvement" to be consistent with the line above.

File: .github/ISSUE_TEMPLATE/documentation_improvement.md (hunk 3 above)
Comment (sarakonradi) [reply .github/ISSUE_TEMPLATE/documentation_improvement.md:19]: It's relevant for installation instructions and documentation workflow matters, but less so for other topics. What do you suggest?

File: .github/ISSUE_TEMPLATE/documentation_improvement.md (hunk 4 above)
Comment (sarakonradi) [reply .github/ISSUE_TEMPLATE/documentation_improvement.md]: Done.

File: .github/ISSUE_TEMPLATE/documentation_improvement.md (hunk 3 above)
Comment (terhorstd) [reply .github/ISSUE_TEMPLATE/documentation_improvement.md:19]: Maybe we can rephrase this to specify the envrionment this is relevant for, e.g. documentation regarding MacOS...

Comment (sarakonradi) [reply .github/ISSUE_TEMPLATE/documentation_improvement.md:19]: Done.

**Comment by sarakonradi** [comment]
@terhorstd @heplesser I updated the ``Desktop/Environment`` section to make it more relevant to documentation-related issues. Please check!

File: .github/ISSUE_TEMPLATE/documentation_improvement.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+---
+name: Documentation improvement
+about: Request a documentation improvement
+title: ''
+labels: 'C: Documentation'
+assignees: ''
+
```
Comment (jougs) [inline .github/ISSUE_TEMPLATE/documentation_improvement.md:7]: Is this empty line needed?

Comment (sarakonradi) [reply .github/ISSUE_TEMPLATE/documentation_improvement.md:7]: @jougs Well-noticed! I added it for consistency with our other issue templates: https://github.com/nest/nest-simulator/tree/master/.github/ISSUE_TEMPLATE

**Pull request merged**